plutus-tx/src/PlutusTx/Blueprint/TH.hs

```diff
@@ -40,10 +45,15 @@ makeHasSchemaInstance dataTypeName indices = do
   let nonOverlapInstance = TH.InstanceD Nothing
 
   -- Lookup indices for all constructors of a data type.
-  indexedCons <- for (TH.datatypeCons dataTypeInfo) $ \ctorInfo ->
-    case lookup (TH.constructorName ctorInfo) indices of
-      Just index -> pure (ctorInfo, index)
-      Nothing    -> fail $ "No index given for constructor " ++ show (TH.constructorName ctorInfo)
+  indexedCons :: [(TH.ConstructorInfo, SchemaInfo, Natural)] <- do
+    typeSchemaInfo <- lookupSchemaInfo dataTypeName
+    for (TH.datatypeCons dataTypeInfo) $ \ctorInfo -> do
+      let ctorName = TH.constructorName ctorInfo
+      case lookup ctorName indices of
+        Nothing -> fail $ "No index given for constructor " ++ show (TH.constructorName ctorInfo)
+        Just index -> do
+          ctorSchemaInfo <- lookupSchemaInfo ctorName
+          pure (ctorInfo, Set.union ctorSchemaInfo typeSchemaInfo, index)
```

Comment:
Is this right? You inherit the annotations from the type onto the constructors? That doesn't seem right to me, and isn't obviously documented. It also IMO behaves wrongly if so: your handling of duplicated annotations is to pick one randomly, but here if anything the constructor annotation should override the type one.

Reply:
I have added a comment that explains this logic.